finder.ts

```diff
@@ -99,34 +107,69 @@ export function finder(input: Element, options?: Partial<Options>): string {
       return selector(fPath)
     }
     count++
-    if (unique(candidate, rootDocument)) {
-      foundPath = candidate
-      break
+    if (unique(candidate, input, rootDocument)) {
+      foundPaths.push(candidate)
+      if (input.length === 1) {
+        break
+      }
```

Comment:
optimization + finding new paths should ideally be done within this search logic

We should also be feeding the found paths into the search iterator so it can short circuit paths that are more expensive.